### PR TITLE
Feature/v2/HintTimer

### PR DESCRIPTION
**Pull request by JonGardner-RedSpace**
Added a simple timer to help with hints in games.

### CODE REVIEW

File: src/hint/HintTimer.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+   */
+  constructor() {
+    this.listeners = [];
+  }
+
+  /**
+   * @param  {Number} time time in milliseconds
+   * @return {void}@memberof HintTimer
+   */
+  start(time) {
```
Comment (SmithAF) [inline src/hint/HintTimer.js]: Would it make sense to just have a default value instead of checking if a value was passed?

Comment (JonGardner-RedSpace) [reply src/hint/HintTimer.js]: I can do that.

File: src/hint/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Hint Timer
+
+The hint timer is a simple resettable timer.
```
Comment (chipbell4) [inline src/hint/README.md]: I think it's worth also providing a use case here. Also, we'll want to link to this readme from the main one in the root folder (with a short description there as well)

Comment (gbhkit) [reply src/hint/README.md]: I second the request for use case/description documentation. SpringRoll 1 has a HintsPlayer class, but we never used it; we just rolled our own hint/prompt logic per game. Curious to know what the idea of the global hint timer and hint button at the Container level are about.

**Comment by JonGardner-RedSpace** [comment]
I made a small breaking change in the constructor of Application that allows devs to set a HintPlayer when the application is created, 

This change will also allow us to add more configs in the future without breaking it again.

File: src/hints/HintSequencePlayer.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * @class HintSequencePlayer
+ */
+export class HintSequencePlayer extends IHintPlayer {
+  /**
+   * Creates an instance of HintSequencePlayer.
+   * @memberof HintSequencePlayer
+   */
+  constructor() {
+    super();
+    /** @type {function()[]} */
```
Comment (SmithAF) [inline src/hints/HintSequencePlayer.js]: What's this doing here?

Comment (JonGardner-RedSpace) [reply src/hints/HintSequencePlayer.js]: mostly for auto complete. it's not strictly necessary.

**Pull request merged**